feat: add jamstack arch (#946)
aziontech/docs: src/content/docs/pt-br/pages/arquiteturas/jamstack/jamstack.mdx

---

> title: Implantando Sites Jamstack na Borda da Rede

Reviewer: ```suggestion
title: Implante Sites Jamstack no Edge da Rede
```

description: Use Azion e Jamstack para melhorar as capacidades dos sites com soluções resilientes, seguras e escaláveis.

> meta_tags: arquitetura, construção, bot, desenvolvimento, jamstack

Reviewer: ```suggestion
meta_tags: arquitetura, construção, desenvolvimento, jamstack
```

namespace: docs_arch_jamstack_deploy

> permalink: /documentacao/arquitetura/jamstack/implantacao-app-jamstack/

Reviewer: ```suggestion
permalink: /documentacao/arquitetura/jamstack/implantacao-aplicacoes-jamstack/
```

---

> As arquiteturas de sites Jamstack utilizam páginas pré-renderizadas e APIs para projetar aplicações web, aprimorando significativamente as capacidades de sites com soluções resilientes, seguras e escaláveis. Essas arquiteturas gerenciam cada parte da aplicação, da interface até o backend, usando APIs em JavaScript, templates de marcação reutilizáveis ​e ativos estáticos pré-construídos, resultando em melhor desempenho, maior segurança, custo-benefício e uma experiência superior para o desenvolvedor. Enquanto isso, a utilização do edge computing - processamento de dados no edge mais proximo do usuário - pode reduzir a latência e o uso de largura de banda, aumentando ainda mais o desempenho.

Reviewer: ```suggestion
As arquiteturas de sites Jamstack utilizam páginas pré-renderizadas e APIs para projetar aplicações web, aprimorando significativamente as capacidades de sites com soluções resilientes, seguras e escaláveis. Essas arquiteturas gerenciam cada parte da aplicação, da interface até o backend, usando APIs em JavaScript, templates de Markup reutilizáveis ​e ativos estáticos pré-construídos, resultando em melhor desempenho, maior segurança, custo-benefício e uma experiência superior para o desenvolvedor. Enquanto isso, a utilização de edge computing - com um processamento de dados no edge, mais proximo do usuário - pode reduzir a latência e o uso de largura de banda, aumentando ainda mais o desempenho.
```

> As arquiteturas de sites Jamstack utilizam páginas pré-renderizadas e APIs para projetar aplicações web, aprimorando significativamente as capacidades de sites com soluções resilientes, seguras e escaláveis. Essas arquiteturas gerenciam cada parte da aplicação, da interface até o backend, usando APIs em JavaScript, templates de marcação reutilizáveis ​e ativos estáticos pré-construídos, resultando em melhor desempenho, maior segurança, custo-benefício e uma experiência superior para o desenvolvedor. Enquanto isso, a utilização do edge computing - processamento de dados no edge da rede mais próxima do usuário - pode reduzir a latência e o uso de largura de banda, aumentando ainda mais o desempenho.

Author: ```suggestion
As arquiteturas de sites Jamstack utilizam páginas pré-renderizadas e APIs para projetar aplicações web, aprimorando significativamente as capacidades de sites com soluções resilientes, seguras e escaláveis. Essas arquiteturas gerenciam cada parte da aplicação, da interface até o backend, usando APIs em JavaScript, templates de marcação reutilizáveis ​e ativos estáticos pré-construídos, resultando em melhor desempenho, maior segurança, custo-benefício e uma experiência superior para o desenvolvedor. Enquanto isso, a utilização do edge computing - processamento de dados no edge mais próxima do usuário - pode reduzir a latência e o uso de largura de banda, aumentando ainda mais o desempenho.
```

> Essa solução é ideal para organizações que buscam melhorar o desempenho do site, melhorar a experiência geral do usuário e capacitar os desenvolvedores a construir e implantar rapidamente aplicativos de alto desempenho e escaláveis no edge da rede, mais próximos dos usuários finais.

Author: ```suggestion
Essa solução é ideal para organizações que buscam melhorar o desempenho do site, melhorar a experiência geral do usuário e capacitar os desenvolvedores a construir e implantar rapidamente aplicações de alto desempenho e escaláveis no edge, mais próximos dos usuários finais.
```

---

## Diagrama de Arquitetura Jamstack

![Diagrama de Arquitetura](/assets/docs/images/uploads/jamstack-deploy.png)

### Fluxo de Dados Jamstack 

> 1. O desenvolvedor interage com a plataforma Azion usando ferramentas como o console baseado na web e a interface de linha de comando (CLI). Essas ferramentas são essenciais para configurar e gerenciar applicações e serviços na plataforma.

Reviewer: ```suggestion
1. O desenvolvedor interage com a plataforma da Azion usando ferramentas como o Console baseado na web e a interface de linha de comando (CLI). Essas ferramentas são essenciais para configurar e gerenciar aplicações e serviços na plataforma.
```

> 2. Os desenvolvedores trabalham no código da aplicação localmente em suas máquinas. Quando prontos, eles confirmam as mudanças no repositório Git local e então enviam esses commits para um repositório remoto. No caso de trabalhar com a Azion, este repositório remoto pode estar vinculado à plataforma Azion para acionar ações subsequentes.

Reviewer: ```suggestion
2. Os desenvolvedores trabalham no código da aplicação localmente em suas máquinas. Quando prontos, eles confirmam as mudanças no repositório Git local e então enviam esses commits para um repositório remoto. No caso de trabalhar com a Azion, este repositório remoto pode estar vinculado à plataforma da Azion para acionar ações subsequentes.
```

> 3. Os arquivos das aplicações são enviados para o Edge Storage, depois é criado um bucket e os arquivos são armazenados nele. O desenvolvedor pode armazenar dados e ativos da aplicação aqui, que podem ser processados rapidamente pelo edge application.

Reviewer: ```suggestion
3. Os arquivos das aplicações são enviados para o Edge Storage, depois é criado um bucket e os arquivos são armazenados nele. O desenvolvedor pode armazenar dados e ativos da aplicação aqui, que podem ser processados rapidamente pela edge application.
```

> 4. Azion Actions configura um fluxo de CI, conectando o repositório GitHub com  Azion criado na implantação. Esse processo é possível através da comunicação entre Azion Actions e Azion API.

Reviewer: ```suggestion
4. Azion Actions configura um fluxo de Continuous Deployment, conectando o repositório GitHub com Azion criado na implantação. Esse processo é possível através da comunicação entre Azion Actions e Azion API.
```

> 4. Azion Actions configura um fluxo de Implantação Contínua, conectando o repositório GitHub com  Azion criado na implantação. Esse processo é possível através da comunicação entre Azion Actions e Azion API.

Author: ```suggestion
4. Azion Actions configura um fluxo de CI, conectando o repositório GitHub com  Azion criado na implantação. Esse processo é possível através da comunicação entre Azion Actions e Azion API.
```

5. Azion Actions implementa  e suas configurações, com base em uma configuração padrão ou na informada pelo usuário em um arquivo manifesto. As configurações incluem Rules Engine, Edge Cache, Image Processor e Edge Functions.

> 6. A saída do edge node é entregue aos usuários finais através de um domínio, representando o ponto de extremidade voltado para o usuário da web onde a aplicação hospedado é acessível.

Reviewer: ```suggestion
6. A saída do edge node é entregue aos usuários finais através de um domínio, representando o endpoint voltado para o usuário da web onde a aplicação hospedado é acessível.
```

### Componentes

- [Azion Console](/pt-br/documentacao/produtos/guias/conhecendo-o-azion-console/): a interface onde você pode acessar todos os produtos disponíveis e suas configurações.

> - [Azion CLI](/pt-br/documentacao/produtos/azion-cli/visao-geral/): uma interface de linha de comando (CLI) de código aberto que permite a interação com a plataforma edge da Azion usando um terminal.

Reviewer: ```suggestion
- [Azion CLI](/pt-br/documentacao/produtos/azion-cli/visao-geral/): uma interface de linha de comando (CLI) de código aberto que permite a interação com a Plataforma Edge da Azion usando um terminal.
```

> - [Edge Application](/pt-br/documentacao/produtos/build/edge-application/): um produto Azion que permite construir aplicações que executam no edge da Azion.

Reviewer: ```suggestion
- [Edge Application](/pt-br/documentacao/produtos/build/edge-application/): um produto Azion que permite construir aplicações que são executadas no edge da Azion.
```

> - [Edge Storage](/pt-br/documentacao/produtos/store/edge-storage/): um serviço de armazenamento escalável e seguro projetado para integrar o armazenamento de objetos com a plataforma edge da Azion usando o padrão S3 para operações de objetos.

Reviewer: ```suggestion
- [Edge Storage](/pt-br/documentacao/produtos/store/edge-storage/): um serviço de armazenamento escalável e seguro projetado para integrar o armazenamento de objetos com a Plataforma Edge da Azion usando o padrão S3 para operações de objetos.
```

---

## Implementação 

> - Inicialize um edge application Jamstack com base em templates:

Author: ```suggestion
- Inicialize uma edge application Jamstack com base em templates:
```

> [Guias específicos de frameworkw](/pt-br/documentacao/produtos/build/develop-with-azion/frameworks-specific/visao-geral/)

Author: ```suggestion
[Guias específicos de frameworkw](/pt-br/documentacao/produtos/build/develop-with-azion/frameworks-specific/visao-geral/)
```
```suggestion
[Guias específicos de framework](/pt-br/documentacao/produtos/build/develop-with-azion/frameworks-specific/visao-geral/)
```
